### DISCUSSION TITLE
Problems of CGAL Clip tools

### DISCUSSION THREAD

**Discussion by Giseribility** [CGAL/cgal#6789] [Q&A] [answered]
When using the PMP Clip tools of CGAL to clip polyhedron with face, the program enters an dead loop or exits abnormally. The face and polyhedron model(.obj format) topologically correct (with no self intersection and normal errors). I wonder what causes the dead loop, the complex triangular surface, or any other predefined conditions which are not satisfied?
![image](https://user-images.githubusercontent.com/46023666/183328745-d3538c52-04a9-4e9a-a758-347cb6745d97.png)
![image](https://user-images.githubusercontent.com/46023666/183328819-98a53989-fb82-4d9a-8c61-5599fe7250cd.png)
![image](https://user-images.githubusercontent.com/46023666/183328309-f6603b5a-def5-4d14-9623-71151a8ad5d4.png)

the  gist URL of testData is as follows:
https://gist.github.com/Giseribility/1fee16c8f61d35d16ab93f1a7174a62f

**Comment by pentacular** [comment]
Hard to tell without a minimal test case and data.

It's also not clear to me what you're trying to clip with what.

**Reply by Giseribility** [reply]
Sry for my unclear expression. I wanner clip the cuboid with the positive camber in Fig.1 and Fig.2. If necessary, I would like to send the obj format data to you privately.

**Reply by Giseribility** [reply]
Test data has been uploaded, please check it.

**Comment by sloriot** [comment]
Please share the data and the code you are using.

**Reply by Giseribility** [reply]
Test data has been uploaded, please check it.
Sry for being late to upload the data,

**Comment by afabri** [comment]
I cannot open the zip file. It seems to be invalid.

**Reply by Giseribility** [reply]
I redownloaded it and opened it succesfully.
Please send me your e-mail if necessary.

**Comment by sloriot** [comment]
Cannot extract the files. Please use gist.github.com

**Reply by Giseribility** [reply]
the gist URL of testData has been uploaded. Please check it.
https://gist.github.com/Giseribility/1fee16c8f61d35d16ab93f1a7174a62f

**Reply by afabri** [reply]
Your mesh has self intersections here
![image](https://user-images.githubusercontent.com/3263539/186134214-1573f985-e096-4362-9d90-1446d6bbeab2.png)

**Reply by Giseribility** [reply]
Wow, thank you very much indeed

**Comment by afabri** [comment, marked as answer]
The mesh has self-intersections.

**Reply by Giseribility** [reply]
Excuse me, what is the model check software used before

**Reply by sloriot** [reply]
[CGAL Polyhedron demo](https://doc.cgal.org/latest/Manual/windows.html#title15): select the mesh and use "Operations->Self-Intersection Test"

**Reply by Giseribility** [reply]
thank you
